### PR TITLE
HEEDLS-425 - Implemented unlocking of admin accounts

### PR DESCRIPTION
**Pull request by DanBloxham-sw**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-425

### Description
Unlock account button on centre admins page now unlocks the account.

### Screenshots
N/A

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by AlexJacksonDS — APPROVED** [review]
Looks good to me.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs [hunk 1]
```diff
@@ -129,6 +129,23 @@ public IActionResult EditAdminRoles(EditRolesViewModel model, int adminId)
             return RedirectToAction("Index");
         }
 
+        [Route("{adminId:int}/UnlockAccount")]
+        [HttpGet]
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs:133]: This feels like it should be a post

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs [hunk 2]
```diff
@@ -129,6 +129,23 @@ public IActionResult EditAdminRoles(EditRolesViewModel model, int adminId)
...
         }
 
+        [Route("{adminId:int}/UnlockAccount")]
+        [HttpGet]
+        public IActionResult UnlockAccount(int adminId)
+        {
+            var centreId = User.GetCentreId();
+            var adminUser = userDataService.GetAdminUserById(adminId);
+
+            if (adminUser == null || adminUser.CentreId != centreId)
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs:139]: Could we write some controller tests for this?

**Review by stellake — APPROVED** [review]
Looks great! 👍

**Pull request merged**